sentry_streams/sentry_streams/examples/spans_buffer.py

```diff
@@ -1,29 +1,93 @@
+import json
+from dataclasses import dataclass
 from datetime import timedelta
+from typing import Any, Self
 
-from sentry_streams.examples.spans import SpansBuffer, build_segment_json, build_span
-from sentry_streams.pipeline.pipeline import (
-    Aggregate,
-    Map,
-    Pipeline,
-    StreamSink,
-    StreamSource,
+from sentry_streams.pipeline import Map, Reducer, StreamSink, streaming_source
+from sentry_streams.pipeline.function_template import (
+    Accumulator,
 )
+from sentry_streams.pipeline.message import Message
 from sentry_streams.pipeline.window import TumblingWindow
 
-pipeline = Pipeline()
 
-source = StreamSource(
-    name="myinput",
-    ctx=pipeline,
-    stream_name="events",
-)
+@dataclass
+class Span:
+    span_id: int
+    trace_id: int
+    duration: int
+    timestamp: int
+
+    def to_dict(self) -> dict[str, int]:
+        return {
+            "span_id": self.span_id,
+            "trace_id": self.trace_id,
+            "duration": self.duration,
+            "timestamp": self.timestamp,
+        }
+
+
+def build_span(value: Message[bytes]) -> Span:
+    """
+    Build a Span object from a JSON str
+    """
+
+    d: dict[str, Any] = json.loads(value.payload)
+
+    return Span(d["span_id"], d["trace_id"], d["duration"], d["timestamp"])
```

Comment:
Let's use the schema from sentry-kafka-schemas and a Parser rather than a custom event type.

Reply:
I thought we were trying to move away from sentry-kafka-schemas?

Reply:
Yes, but the replacement will have an interface that will be very close to that one. It should be a quick swap, we can use it till we replace it.